mlrun/artifacts/document.py

```diff
@@ -104,20 +104,20 @@ def __new__(
         class DynamicDocumentLoader(BaseLoader):
```

Comment:
I have to say I still don't get the reasoning for using an internal-class here and why this code (the `DynamicDocumentLoader::__new__` method) cannot simply be implemented in the `MLRunLoader::__init__` method. Of course, this means that the `MLRunLoader` class will inherit from `BaseLoader` directly.

Reply:
We cannot inherit from BaseLoader directly, because when langchain is not installed it will generate errors